packs/elastic/actions/search.py

```diff
@@ -0,0 +1,87 @@
+from easydict import EasyDict
+from lib.items_selector import ItemsSelector
+from lib.elastic_action import ElasticAction
+from collections import defaultdict
+import logging
+import sys
+import elasticsearch
+import json
+
+logger = logging.getLogger(__name__)
+
+
+class SearchRunner(ElasticAction):
+
+    def __init__(self, config=None):
+        super(SearchRunner, self).__init__(config=config)
+        self._iselector = None
+
+
+    @property
+    def iselector(self):
+        """
+        Used to fetch indices/snapshots and apply filter to them.
+        """
+        if not self._iselector:
+            self._iselector = ItemsSelector(self.client, **self.config)
+        return self._iselector
+
+
+    def run(self, **kwargs):
```

Comment:
Hmm, got it interesting idea. I will uncouple than, though I don't prefer having different class for micro portions of code (in case with search). But following this idea it would be better to uncouple +1